src/vivarium_gates_nutrition_optimization/components/children.py

```diff
@@ -1,11 +1,144 @@
 from pathlib import Path
+from typing import Tuple
 
+import numpy as np
 import pandas as pd
 from vivarium.framework.engine import Builder
 from vivarium.framework.event import Event
 from vivarium_cluster_tools.utilities import mkdir
 
-from vivarium_gates_nutrition_optimization.constants import models
+from vivarium_gates_nutrition_optimization.constants import (
+    data_keys,
+    data_values,
+    models,
+)
+
+
+class NewChildren:
+    def __init__(self):
+        self.lbwsg = LBWSGDistribution()
+
+    @property
+    def name(self):
+        return "child_status"
+
+    @property
+    def sub_components(self):
+        return [self.lbwsg]
+
+    @property
+    def columns_created(self):
+        return ["sex_of_child", "birth_weight", "gestational_age"]
+
+    def setup(self, builder: Builder):
+        self.randomness = builder.randomness.get_stream(self.name)
+        self.male_sex_percentage = data_values.INFANT_MALE_PERCENTAGES[
+            builder.data.load(data_keys.POPULATION.LOCATION)
+        ]
+
+    def empty(self, index: pd.Index) -> pd.DataFrame:
+        return pd.DataFrame(
+            {
+                "sex_of_child": models.INVALID_OUTCOME,
+                "birth_weight": np.nan,
+                "gestational_age": np.nan,
+            },
+            index=index,
+        )
+
+    def __call__(self, index: pd.Index):
```

Comment:
Why did you make this callable rather than defining a function like `generate_children()`?